tests/components/konnected/test_panel.py

```diff
@@ -0,0 +1,375 @@
+                    "pause": 100,
+                    "repeat": 4,
+                    "state": None,
+                    "zone": "out",
+                },
+                {
+                    "activation": "high",
+                    "momentary": None,
+                    "name": "Konnected 445566 Actuator 6",
+                    "pause": None,
+                    "repeat": None,
+                    "state": None,
+                    "zone": "6",
+                },
+            ],
+        }
+    }
+
+
+async def test_create_and_setup_pro(hass, mock_panel):
+    """Test that we create a Konnected Pro Panel and save the data."""
+    device_config = config_flow.CONFIG_ENTRY_SCHEMA(
+        {
+            "host": "1.2.3.4",
+            "port": 1234,
+            "id": "112233445566",
+            "model": "Konnected Pro",
+            "access_token": "11223344556677889900",
+            "default_options": config_flow.OPTIONS_SCHEMA({config_flow.CONF_IO: {}}),
+        }
+    )
+
+    device_options = config_flow.OPTIONS_SCHEMA(
+        {
+            "io": {
+                "2": "Binary Sensor",
+                "6": "Binary Sensor",
+                "10": "Binary Sensor",
+                "3": "Digital Sensor",
+                "7": "Digital Sensor",
+                "11": "Digital Sensor",
+                "4": "Switchable Output",
+                "8": "Switchable Output",
+                "out1": "Switchable Output",
+                "alarm1": "Switchable Output",
+            },
+            "binary_sensors": [
+                {"zone": "2", "type": "door"},
+                {"zone": "6", "type": "window", "name": "winder", "inverse": True},
+                {"zone": "10", "type": "door"},
+            ],
+            "sensors": [
+                {"zone": "3", "type": "dht"},
+                {"zone": "7", "type": "ds18b20", "name": "temper"},
+                {"zone": "11", "type": "dht", "poll_interval": 5},
+            ],
+            "switches": [
+                {"zone": "4"},
+                {
+                    "zone": "8",
+                    "name": "switcher",
+                    "activation": "low",
+                    "momentary": 50,
+                    "pause": 100,
+                    "repeat": 4,
+                },
+                {"zone": "out1"},
+                {"zone": "alarm1"},
+            ],
+        }
+    )
+
+    entry = MockConfigEntry(
+        domain="konnected",
+        title="Konnected Pro Alarm Panel",
+        data=device_config,
+        options=device_options,
+    )
+    entry.add_to_hass(hass)
+    hass.data[panel.DOMAIN] = {
+        panel.CONF_API_HOST: "192.168.1.1",
+    }
+
+    device = panel.AlarmPanel(hass, entry)
+    await device.async_save_data()
+    await device.async_connect()
+    await device.update_switch("2", 1)
+
+    # confirm the correct api is used
+    # pylint: disable=no-member
+    assert device.client.put_device.call_count == 0
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4986898)